### PR TITLE
Improve documentation structure

### PR DESCRIPTION
**Pull request by alachambre**
Massive update on the taxonomy to improve the documentation structure

- fewer top level entries, with understanding thematics
- Index pages with cards for nodes

### CODE REVIEW

**Comment by tbouffard** [comment]
⚠️ About the new page names: they are not consistent with the existing pages (we are not using camel case).
This mainly impacts the urls.

File: modules/ROOT/taxonomy.adoc [hunk 1]
```diff
@@ -1,340 +1,243 @@
...
+ ** xref:lifecycle-and-profiles.adoc[Project phases]
+ ** xref:design-methodology.adoc[Conception]
+ ** xref:workspaces-and-repositories.adoc[Workspaces and projects]
+ ** xref:shared-project.adoc[Set up Bonita Studio with GitHub]
+ ** xref:share-a-repository-on-github.adoc[Share a project on GitHub]
+ ** xref:automating-builds.adoc[Automate builds]
+ ** xref:set-up-continuous-integration.adoc[Set up continuous integration]
+* Automation project
+ ** Project artefacts
+  *** Core artifacts
```
Comment (educhastenier) [inline modules/ROOT/taxonomy.adoc]: ⚠️ empty section

File: modules/ROOT/taxonomy.adoc [hunk 2]
```diff
@@ -1,340 +1,243 @@
...
+  *** Extensions
```
Comment (educhastenier) [inline modules/ROOT/taxonomy.adoc]: ⚠️ empty section

File: old_taxonomy.adoc [hunk 3]
```diff
@@ -0,0 +1,352 @@
+* xref:what-is-bonita.adoc[What is Bonita?]
```
Comment (educhastenier) [inline old_taxonomy.adoc:1]: This file should be removed before merging this PR

File: modules/ROOT/taxonomy.adoc [hunk 4]
```diff
@@ -350,3 +273,23 @@
...
+ ** xref:use-gzip-compression.adoc[Using gzip compression]
+* xref:bonita-version-update-index.adoc[Bonita version update]
+ ** xref:product-versioning.adoc[Bonita versions]
+ ** xref:migration-overview.adoc[Update overview]
+ ** xref:migrate-from-an-earlier-version-of-bonita-bpm.adoc[Migrate from an earlier version of Bonita]
+  *** xref:migration-tool.adoc[Migration tool changelog]
+ ** xref:migrate-a-form-from-6-x.adoc[Migrate a form from 6.x]
+* xref:building-community-edition-from-source.adoc[Contributing to Bonita]
+* xref:how-tos-index.adoc[How tos]
+* xref:troubleshooting.adoc[Troubleshooting]
```
Comment (educhastenier) [inline modules/ROOT/taxonomy.adoc]: ⚠️ Many conflicts to come when merging 2021.1 into 2021.2 ⚠️

File: modules/ROOT/pages/bonita-version-update-index.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
+= Bonita version update 
+:description: blabla
```
Comment (educhastenier) [inline modules/ROOT/pages/bonita-version-update-index.adoc:2]: ⚠️ 30 occurrences of **blabla** in this PR !!!
😲

**Pull request merged**